.github/workflows/update-pinned-reqs.yml

```diff
@@ -23,14 +23,8 @@ jobs:
       with:
         python-version: 3.12
 
-    - name: Upgrade pip
-      run: python -m pip install --upgrade pip
-
-    - name: Install pip-tools
-      run: python -m pip install pip-tools
-
-    - name: Generate requirements.txt
-      run: pip-compile --output-file=requirements.txt --rebuild --upgrade --all-extras
+    - name: Rebuild requirements
+      run: tox -e requirements
```

Comment:
Creating a tox environment for rebuilding requirements makes it a lot more straightforward to rebuild requirements locally with `tox -e requirements` in the same way that is done in CI.